interp/run.go

```diff
@@ -466,13 +468,10 @@ func firstMissingMethod(src, dest reflect.Type) string {
 func convert(n *node) {
 	dest := genValue(n)
 	c := n.child[1]
-	typ := n.child[0].typ.TypeOf()
+	typ := n.child[0].typ.frameType()
 	next := getExec(n.tnext)
 
 	if c.isNil() { // convert nil to type
-		if n.child[0].typ.cat == interfaceT {
-			typ = reflect.TypeOf((*valueInterface)(nil)).Elem()
-		}
```

Comment:
I have not thought particularly hard on this change, but it seemed it could be done since frameType() should now give us the correct typ in the case of an interfaceT. And there does not seem to be any regression in the tests. But if it looks suspicious to you I can think harder about it.